contracts/external/cw-token-swap/src/contract.rs

```diff
@@ -112,14 +114,17 @@ fn do_fund(
     storage.save(deps.storage, &counterparty)?;
 
     let messages = if counterparty.provided && other_counterparty.provided {
-        vec![
-            counterparty
-                .promise
-                .into_send_message(&other_counterparty.address)?,
-            other_counterparty
+        let mut msgs = counterparty
+            .promise
+            .clone()
+            .into_send_message(other_counterparty.address.to_string(), false)?;
+
+        msgs.append(
+            &mut other_counterparty
                 .promise
-                .into_send_message(&counterparty.address)?,
-        ]
+                .into_send_message(counterparty.address.to_string(), false)?,
+        );
+        msgs
```

Comment:
why rewrite this? seems like we have an additional clone this way.

Reply:
Initially `into_send_message` was returning a cosmosMsg, but our change makes it that `into_send_message` returns a **vector** of cosmosMsgs, so we need to combine the vectors, rather then create new vector from 2 msgs.

Changed it to use `extend` instead of `append` which removes the `&mut from the other_counterparty vector.

For some reason I managed to remove the clone, not sure why I had it in the first place tho, seems not related to the `extend` move.